lib/datasets/rdatasets.rb

```diff
@@ -0,0 +1,109 @@
+require_relative "dataset"
+require_relative "tar_gz_readable"
+
+module Datasets
+  class RdatasetsList < Dataset
+    Record = Struct.new(:package,
+                        :dataset,
+                        :title,
+                        :rows,
+                        :cols,
+                        :n_binary,
+                        :n_character,
+                        :n_factor,
+                        :n_logical,
+                        :n_numeric,
+                        :csv,
+                        :doc)
+
+    def initialize
+      super
+      @metadata.id = "rdatasets"
+      @metadata.name = "Rdatasets"
+      @metadata.url = "https://vincentarelbundock.github.io/Rdatasets/"
+      @metadata.licenses = ["GPL-3"]
+      @data_url = "https://raw.githubusercontent.com/vincentarelbundock/Rdatasets/master/datasets.csv"
+      @data_path = cache_dir_path + "datasets.csv"
+    end
+
+    def download(force: false)
+      if force || !@data_path.exist?
+        super(@data_path, @data_url)
+      end
+    end
+
+    def dataset_info(package_name, dataset_name)
+      each_dataset(package_name) do |row|
+        if row["Item"] == dataset_name
+          record = row.to_h
+          record.transform_keys! {|key| key.downcase.to_sym }
+          return record
+        end
+      end
+    end
+
+    def each(package_name = nil)
```

Comment:
Do we need `package_name` API?

Is `rdataset_list.select {|rd| rd.package == package_name}` a popular task?

Reply:
I don't know whether such a task occurs frequently.
[Rdatasets.jl](https://github.com/JuliaStats/RDatasets.jl) provides [such a feature](https://github.com/JuliaStats/RDatasets.jl/blob/master/src/datasets.jl#L1-L4).
I follow it.

Reply:
OK.
If we add support for filtering feature, I want to implement it as more generic/flexible/extensible API such as `dataset.filter(package: "datasets").each` (`filter` can be used by all datasets) and `dataset.where(package: "datasets").limit(10).each` (Active Record like style).

How about removing `package_name = nil` in pull request and consider API as another task?

Reply:
Sounds reasonable. I'll do it.

Reply:
Added `RdatasetList#filter` at 3dd3a99.